napari/_qt/_tests/test_qt_viewer.py

```diff
@@ -675,3 +678,67 @@ def test_create_non_empty_viewer_model(qtbot):
     del viewer
     qtbot.wait(50)
     gc.collect()
+
+
+def test_async_slice_image_on_current_step_change(make_napari_viewer, qtbot):
```

Comment:
These tests illustrate the integration that we've added in this PR. Maybe they should be in a special test module?